pywatts/modules/generation/anomaly_generation_module.py

```diff
@@ -0,0 +1,390 @@
+        :type anomaly: str
+        :param anomaly_params: Json dict containing anomaly method parameters (default {}).
+        :type anomaly_params: Dict
+        :param length_params: Json dict containing length distribution parameters (default {}).
+        :type length_params: Dict
+        :param label: Label to use for the anomaly labels (default None).
+        :type label: Optional[int]
+        :param seed: Seed to be used by the random generator (default 0).
+        :type seed: int
+        """
+        super().__init__(name)
+        self.count = count
+        self._check_anomaly_type(anomaly)
+        self.anomaly = anomaly
+        self.anomaly_params = anomaly_params
+        self.length_params = length_params
+        self.label = label
+        self.seed = seed
+
+    def get_params(self) -> Dict[str, object]:
+        """
+        Get parameters of the anomaly generation module as Dict.
+
+        :return: Dict containing all parameters.
+        :rtype: Dict[str, object]
+        """
+        return {
+            "count": self.count,
+            "anomaly": self.anomaly,
+            "anomaly_params": self.anomaly_params,
+            "length_params": self.length_params,
+            "seed": self.seed,
+        }
+
+    def set_params(self, count: Optional[Union[int, float]] = None,
+                   anomaly: Optional[str] = None, anomaly_params: Optional[Dict] = None,
+                   length_params: Optional[Dict] = None, label: Optional[str] = None,
+                   seed: Optional[int] = None):
+        """
+        Set parameters of the anomaly generation module.
+
+        :param count: Number of anomalies to be inserted.
+                      It can be a percentage value (float) or a whole number (int).
+        :type count: Optional[Union[int, float]]
+        :param anomaly: Type of anomaly to be inserted, e.g. 'gap', 'outlier', 'negate' or 'constant.
+        :type anomaly: Optional[str]
+        :param anomaly_params: JSON Dict containing anomaly method parameters.
+        :type anomaly_params: Optional[Dict]
+        :param length_params: JSON Dict containing length distribution parameters.
+        :type length_params: Optional[Dict]
+        :param label: Label to use for the anomaly labels (default None).
+        :type label: Optional[int]
+        :param seed: Seed to be used by the random generator.
+        :type seed: Optional[int]
+        """
+        if count is not None:
+            self.count = count
+        if anomaly is not None:
+            self._check_anomaly_type(anomaly)
+            self.anomaly = anomaly
+        if anomaly_params is not None:
+            self.anomaly_params = anomaly_params
+        if length_params is not None:
+            self.length_params = length_params
+        if label is not None:
+            self.label = label
+        if seed is not None:
+            self.seed = seed
+
+    def _check_anomaly_type(self, anomaly: str):
```

Comment:
Probably this would not needed if you use enums

Reply:
I can update that method to check for all available anomaly types by filtering for all _anomaly_ methods? I added this method to notify the user as early as possible that the anomaly type is not available.